Move to connection object based API

martinpitt opened the pull request:
Keep the `connection` and underlying `cockpit.http()` object for each user, and change the client.js API to take a connection object instead of an uid.

With that we don't have to create a new HTTP connection for each request (thus more efficient), and gain the possibility to close a running monitor of a particular user (which will become important for multi-user support).

---

Part of https://issues.redhat.com/browse/COCKPIT-1086

cockpituous commented on src/PodActions.jsx:
```diff
@@ -90,7 +90,7 @@ export const PodActions = ({ onAddNotification, pod }) => {
                               className="pod-action-force-stop"
                               onClick={() =>
-                                  client.postPod(pod.uid, "stop", pod.Id, { t: 0 })
+                                  client.postPod(con, "stop", pod.Id, { t: 0 })
```
This added line is not executed by any test.

cockpituous replied:
This added line is not executed by any test.

cockpituous replied:
This added line is not executed by any test.

cockpituous replied:
This added line is not executed by any test.

cockpituous commented on src/PodActions.jsx:
```diff
@@ -112,7 +112,7 @@ export const PodActions = ({ onAddNotification, pod }) => {
                               className="pod-action-force-restart"
                               onClick={() =>
-                                  client.postPod(pod.uid, "restart", pod.Id, { t: 0 })
+                                  client.postPod(con, "restart", pod.Id, { t: 0 })
```
This added line is not executed by any test.

cockpituous replied:
This added line is not executed by any test.

cockpituous replied:
This added line is not executed by any test.

cockpituous replied:
This added line is not executed by any test.

cockpituous commented on src/app.jsx:
```diff
@@ -176,159 +176,159 @@ class Application extends React.Component {
+                    this.updateContainerStats(con);
                 })
-                .catch(e => console.warn("initContainers uid", uid, "getContainers failed:", e.toString()));
+                .catch(e => console.warn("initContainers uid", con.uid, "getContainers failed:", e.toString()));
```
This added line is not executed by any test.

cockpituous replied:
This added line is not executed by any test.

cockpituous commented on src/app.jsx:
```diff
@@ -176,159 +176,159 @@ class Application extends React.Component {
                 })
                 .catch(ex => {
-                    console.warn("Failed to do updatePods for uid", uid, ":", JSON.stringify(ex));
+                    console.warn("Failed to do updatePods for uid", con.uid, ":", JSON.stringify(ex));
```
This added line is not executed by any test.

cockpituous replied:
This added line is not executed by any test.

cockpituous commented on src/app.jsx:
```diff
@@ -176,159 +176,159 @@ class Application extends React.Component {
                 })
                 .catch(ex => {
-                    console.warn("Failed to do updatePod for uid", uid, ":", JSON.stringify(ex));
+                    console.warn("Failed to do updatePod for uid", con.uid, ":", JSON.stringify(ex));
```
This added line is not executed by any test.

cockpituous replied:
This added line is not executed by any test.

martinpitt commented:
@jelly I finally have a multi-user demo which disconnects "third party" users again after switching away from them. So this is good! :+1: (Establishing this took a while, sorry)

martinpitt commented:
This got conflicted, resolved.

martinpitt commented:
Fixed a [race condition](https://cockpit-logs.us-east-1.linodeobjects.com/pull-2024-5452abe8-20250227-095233-rhel-10-0/log.html) that I introduced. Not sure why it didn't hit the tests here, but it is rather pronounced in #2024.

cockpituous commented on src/app.jsx:
```diff
@@ -174,159 +174,159 @@ class Application extends React.Component {
+                    this.updateContainerStats(con);
                 })
-                .catch(e => console.warn("initContainers uid", uid, "getContainers failed:", e.toString()));
+                .catch(e => console.warn("initContainers uid", con.uid, "getContainers failed:", e.toString()));
```
This added line is not executed by any test.

cockpituous replied:
This added line is not executed by any test.

cockpituous commented on src/app.jsx:
```diff
@@ -174,159 +174,159 @@ class Application extends React.Component {
                 })
                 .catch(ex => {
-                    console.warn("Failed to do updatePods for uid", uid, ":", JSON.stringify(ex));
+                    console.warn("Failed to do updatePods for uid", con.uid, ":", JSON.stringify(ex));
```
This added line is not executed by any test.

cockpituous replied:
This added line is not executed by any test.

cockpituous commented on src/app.jsx:
```diff
@@ -174,159 +174,159 @@ class Application extends React.Component {
                 })
                 .catch(ex => {
-                    console.warn("Failed to do updatePod for uid", uid, ":", JSON.stringify(ex));
+                    console.warn("Failed to do updatePod for uid", con.uid, ":", JSON.stringify(ex));
```
This added line is not executed by any test.

cockpituous replied:
This added line is not executed by any test.

martinpitt commented:
Unconflicted, and I fixed the silly `con_for() declaration in src/Images.jsx (it was previously too deeply nested)

cockpituous commented on src/app.jsx:
```diff
@@ -174,159 +174,159 @@ class Application extends React.Component {
                 })
                 .catch(ex => {
-                    console.warn("Failed to do updateImages for uid", uid, ":", JSON.stringify(ex));
+                    console.warn("Failed to do updateImages for uid", con.uid, ":", JSON.stringify(ex));
```
This added line is not executed by any test.
